src/client.js

```diff
@@ -252,17 +245,8 @@ assign(Client.prototype, {
 
   // Destroy the current connection pool for the client.
   destroy(callback) {
-    const promise = new Promise((resolver) => {
-      if (!this.pool) {
-        return resolver()
-      }
-      this.pool.drain(() => {
-        this.pool.destroyAllNow(() => {
-          this.pool = undefined
-          resolver()
-        })
-      })
-    })
+    const promise = this.pool ? this.pool.destroy() : Promise.resolve()
```

Comment:
This is now fixed in Tarn 0.1.5. I updated the package.json in the pull request.